app/gui2/src/components/visualizations/ScatterplotVisualization.vue

```diff
@@ -255,8 +260,8 @@ const symbol: d3.Symbol<unknown, Point> = d3.symbol()
 
 const animationDuration = computed(() => (shouldAnimate.value ? ANIMATION_DURATION_MS : 0))
 const margin = computed(() => {
-  const xLabel = data.value.axis.x.label
-  const yLabel = data.value.axis.y.label
+  const xLabel = data.value.axis.x?.label
+  const yLabel = data.value.axis.y?.label
```

Comment:
Are the type declarations accurate? If so I don't see why option chaining would be necessary here

Reply:
This was her as when there was incorrect data for the scatterplot it was erroring here and causing odd behaviour, I could do a different type check if that's better?

Reply:
I'm just trying to understand why we need a check that according to the type declarations shouldn't be necessary. So is the problem that the `props.data` received may contain an `axis` field that doesn't have both `x` and `y`? If so, it seems to me:
- If this is not supposed to happen, can't we fix the preprocessor producing the data?
- If this is an expected case, we should change `interface Data` to have `axis: AxesConfiguration`, and then handle this case in the `const data` computed value.

Reply:
I have added the check to when the data is computed